master/buildbot/changes/bitbucket.py

```diff
@@ -14,184 +14,282 @@
 
-_UNSPECIFIED = object()
+log = Logger()
 
+BASE_URL = "https://api.bitbucket.org"
+OAUTH_URL = 'https://bitbucket.org/site/oauth2/access_token'
+_GET_TOKEN_DATA = {
+    'grant_type': 'client_credentials'
+}
 
-class BitbucketPullrequestPoller(base.PollingChangeSource):
 
-    compare_attrs = ("owner", "slug", "branch",
-                     "pollInterval", "useTimestamps",
-                     "category", "project", "pollAtLaunch")
+class BitbucketPullrequestPoller(base.ReconfigurablePollingChangeSource,
+                                 StateMixin):
 
-    db_class_name = 'BitbucketPullrequestPoller'
+    compare_attrs = ("owner", "slug", "branches", "pollInterval",
+                     "category", "pollAtLaunch", "name")
+    db_class_name = "BitBuckerPullrequestPoller"
 
-    def __init__(self, owner, slug,
-                 branch=None,
-                 pollInterval=10 * 60,
-                 useTimestamps=True,
-                 category=None,
-                 project='',
-                 pullrequest_filter=True,
-                 encoding=_UNSPECIFIED,
-                 pollAtLaunch=False
-                 ):
+    def __init__(self, owner, slug, **kwargs):
+        name = kwargs.get("name")
+        if not name:
+            kwargs["name"] = "BitbucketPullrequestPoller:" + owner + "/" + slug
+        super(BitbucketPullrequestPoller, self).__init__(owner, slug, **kwargs)
+        self.debug = kwargs.get("debug")
+        self.verify = None
+        self.codebase = kwargs.get('codebase')
+
+    def checkConfig(self,
+                    owner,
+                    slug,
+                    branches=None,
+                    pollInterval=10 * 60,
+                    pollAtLaunch=False,
+                    category='pull',
+                    pullrequest_filter=True,
+                    codebase='',
+                    oauth_key=None,
+                    oauth_secret=None,
+                    repository_type="https",
+                    debug=False,
+                    **kwargs):
+        if repository_type not in ["https", "ssh"]:
+            config.error(
+                "repository_type must be one of {https, svn, git, ssh}")
+        super().checkConfig(name=self.name, **kwargs)
+
+    @defer.inlineCallbacks
+    def reconfigService(self,
+                        owner,
+                        slug,
+                        branches=None,
+                        pollInterval=10 * 60,
+                        pollAtLaunch=False,
+                        category='pull',
+                        pullrequest_filter=True,
+                        codebase='',
+                        oauth_key=None,
+                        oauth_secret=None,
+                        debug=False,
+                        repository_type="https",
+                        **kwargs):
+
+        yield super().reconfigService(name=self.name, **kwargs)
+
+        http_headers = {'User-Agent': 'Buildbot'}
```

Comment:
are you sure that the set of headers is not going to increase?  if not, please re-format as
```
http_headers = {
    'User-Agent': 'Buildbot'.
}
```